opentelemetry/proto/profiles/v1development/profiles.proto

```diff
@@ -454,21 +454,21 @@ message Line {
   // Reference to function in Profile.function_table.
   int32 function_index = 1;
   // Line number in source code.
-  int64 line = 2;
+  optional int64 line = 2;
```

Comment:
Line and column are 1-based, so we could easily say here "0 means unset" I think? Not sure if optional buys us much here. I do like the purity and the cleanness of optional, but all the potential quirks and potential performance penalty of using it does give me a pause.

Reply:
"0 means unset" sounds also good to me. Will update it.

Reply:
They're 1-indexed in DWARF but I'm not sure if that's the case for every interpreted language we support. I'd expect line numbers to be 1-indexed across the board but a quick search for column numbers in Python shows 0-indexing:

https://docs.python.org/3/library/traceback.html#traceback.FrameSummary.colno

Reply:
That example is for columns, maybe we could say that the column number is only valid if the line number is non-zero and assume line numbers are 1-indexed? I would hope that the line indexation is always 1-based and seems true for that example above as well, it says

> end_lineno[¶](https://docs.python.org/3/library/traceback.html#traceback.FrameSummary.end_lineno)
The last line number of the source code for this frame. By default, it is set to lineno and indexation starts from 1.

(interestingly there is no this clarifying comment there for the `lineno` field, but I assume it's the same, 1-based)